propagators/awsxray/awsxray_propagator.go

```diff
@@ -0,0 +1,180 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aws
+
+import (
+	"context"
+	"errors"
+	"strings"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+const (
+	traceHeaderKey       = "X-Amzn-Trace-Id"
+	traceHeaderDelimiter = ";"
+	kvDelimiter          = "="
+	traceIdKey           = "Root"
+	sampleFlagKey        = "Sampled"
+	parentIdKey          = "Parent"
+	traceIdVersion       = "1"
+	traceIdDelimiter     = "-"
+	isSampled            = "1"
+	notSampled           = "0"
+
+	traceFlagNone           = 0x0
+	traceFlagSampled        = 0x1 << 0
+	traceIdLength           = 35
+	traceIdDelimitterIndex1 = 1
+	traceIdDelimitterIndex2 = 10
+	traceIdFirstPartLength  = 8
+	parentIdLength          = 16
+	sampledFlagLength       = 1
+)
+
+var (
+	empty                  = trace.EmptySpanContext()
+	errInvalidTraceHeader  = errors.New("invalid X-Amzn-Trace-Id header value, should contain 3 different part separated by ;")
+	errMalformedTraceID    = errors.New("cannot decode trace id from header, should be a string of hex, lowercase trace id can't be all zero")
+	errInvalidSpanIDLength = errors.New("invalid span id length, must be 16")
+)
+
+// AwsXray propagator serializes Span Context to/from AWS X-Ray headers
+//
+// AWS X-Ray format
+//
+// X-Amzn-Trace-Id: Root={traceId};Parent={parentId};Sampled={samplingFlag}
+type AwsXray struct{}
+
+var _ otel.TextMapPropagator = &AwsXray{}
```

Comment:
what declaration does this do/accomplish?

Reply:
I think this statement is a compile time assertion to check that `AwsXray` implements the `otel.TextMapPropagator` interface. I found a similar check being done in the [aws.go](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/master/detectors/aws/aws.go#L43).

Reply:
@KKelvinLo could you add a comment explaining what this line does for future developers?

Reply:
Done, good point